### PR TITLE
Fix for elbow glitches

### PR DESCRIPTION
**Pull request by hyperlogic**
The pole vector constraint calculation within the IK system would sometimes compute the incorrect rotations. This would be visible as an instantaneous snap of the elbow joint as the bicep was curled.

When applying pole vector constraints, there needs to be two methods of determining the current orientation of the elbow joint. One for when the arm/elbow joint is bent, and one for when the arm/elbow is straight. Previously, the way we would switch between these two solutions could cause a large rotation delta to occur between very small angles. Now we use the more accurate method (1) more often, and we smoothly blend between the solutions as the joint gets straighter.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6237/app_12154_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6237/HighFidelity-Beta-PR12154-467a7eaf2dbc47245daf49299a014f8d5a68fda0.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6237/HighFidelity-Beta-PR12154-467a7eaf2dbc47245daf49299a014f8d5a68fda0.dmg

**Comment by hyperlogic** [comment]
QA Test plan:

1. Download the following input recording. [blue-man-shoulder-glitch.json.gz](https://s3.amazonaws.com/hifi-public/tony/input-recodings/blue-man-shoulder-glitch.json.gz)
2. Load the following script. `scripts/developer/inputRecording.js`
3. Press the IRecord button on the toolbar.
4. Press load recording, and browse to the blue-man-shoulder-glitch recording.
5. Enter third person view, or mirror view.  Press play recording.
6. The right elbow should not pop as the bicep is curled.

This is somewhat subjective but, in general, the elbows of avatars should be more stable after this PR.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6245/app_12154_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6245/HighFidelity-Beta-PR12154-983e1fe0759e914de44f1d18f99f658b04afb249.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6245/HighFidelity-Beta-PR12154-983e1fe0759e914de44f1d18f99f658b04afb249.exe

**Comment by RyanDowne** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6256/app_12154_PR.apk

File: libraries/animation/src/AnimInverseKinematics.cpp [hunk 1]
```diff
@@ -636,14 +648,26 @@ void AnimInverseKinematics::solveTargetWithCCD(const AnimContext& context, const
 
                     glm::vec3 dUnit = d / dLen;
```
Comment (AndrewMeadows) [inline libraries/animation/src/AnimInverseKinematics.cpp]: Other than the new warning about how `dUnit` is created but never used in this context: the code looks good.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6256/HighFidelity-Beta-PR12154-983e1fe0759e914de44f1d18f99f658b04afb249.exe

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6261/app_12154_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6261/HighFidelity-Beta-PR12154-bfd18e3696678e71f97dedd62ca4724ac9baa351.exe

**Comment by sbennett77** [comment]
Passed.

@conklin94122

**Pull request merged**